tests/SpecTests/Context.php

```diff
@@ -312,13 +327,17 @@ public function prepareOptions(array $options)
                 throw new LogicException('Unsupported writeConcern args: ' . implode(',', array_keys($diff)));
             }
 
-            $w = $writeConcern['w'];
-            $wtimeout = $writeConcern['wtimeout'] ?? 0;
-            $j = $writeConcern['j'] ?? null;
+            if (! empty($writeConcern)) {
+                $w = $writeConcern['w'];
+                $wtimeout = $writeConcern['wtimeout'] ?? 0;
+                $j = $writeConcern['j'] ?? null;
 
-            $options['writeConcern'] = isset($j)
-                ? new WriteConcern($w, $wtimeout, $j)
-                : new WriteConcern($w, $wtimeout);
+                $options['writeConcern'] = isset($j)
+                    ? new WriteConcern($w, $wtimeout, $j)
+                    : new WriteConcern($w, $wtimeout);
+            } else {
+                unset($options['writeConcern']);
+            }
```

Comment:
This block does not seem to handle additional fields existing in `$writeConcern`. If a spec test has any such fields, they would be silently ignored instead of producing an error because the spec runner does not handle them.

Reply:
The code block a couple of lines above handles this:
```
$writeConcern = (array) $options['writeConcern'];
$diff = array_diff_key($writeConcern, ['w' => 1, 'wtimeout' => 1, 'j' => 1]);

if (! empty($diff)) {
    throw new LogicException('Unsupported writeConcern args: ' . implode(',', array_keys($diff)));
}
```